lib/floe/workflow/choice_rule/data.rb

```diff
@@ -4,6 +4,17 @@ module Floe
   class Workflow
     class ChoiceRule
       class Data < Floe::Workflow::ChoiceRule
+        COMPARE_KEYS = %w[IsNull IsPresent IsNumeric IsString IsBoolean IsTimestamp String Numeric Boolean Timestamp].freeze
+
+        attr_reader :compare_key
+
+        def initialize(*)
+          super
+          @compare_key = payload.keys.detect { |key| key.match?(/^(#{COMPARE_KEYS.join("|")})/) }
```

Comment:
TODO I don't like the regular expression matcher here because it actually matches on typos like "StringEqualz" (this was my first spec).  This is fine when we have the big `case compare_key` in `true?` since the else raises an exception but I want it to be stricter in the payload initialize validation.